app/Http/Controllers/Api/PermissionController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\Api\UpdateRolePermissionsRequest;
+use App\Http\Requests\Api\UpdateUserRolesRequest;
+use App\Models\User;
+use Spatie\Permission\Models\Role;
+
+class PermissionController extends Controller
+{
+    // Update Role Permission
+    public function updateRolePermissions(UpdateRolePermissionsRequest $request)
+    {
+        $validatedData = $request->validated();
+        if (! isset($validatedData['permissions'])) {
+            return response()->json(['status' => 'Permissions is required']);
+        }
+        $role = Role::find($validatedData['roleID']);
+        $permissions = array_pluck($validatedData['permissions'], 'id');
+        $isUpdated = $role->syncPermissions($permissions);
+
+        return response()->json(['status' => 'Role Permission updated successfully']);
+    }
+
+    // Update the User roles
+    public function updateUserRoles(UpdateUserRolesRequest $request)
```

Comment:
this is incorrect place to have this function. The functionality to update roles for a user should be in the UserController. `UserController` is the place where any action related to user should be put in.